This custom strptime() conflicts with strptime() provided by the system in time.h.
--- hexchat-2.9.5/src/fe-gtk/banlist.c.orig	2013-09-08 13:20:35.000000000 +0200
+++ hexchat-2.9.5/src/fe-gtk/banlist.c	2013-09-08 13:20:52.000000000 +0200
@@ -583,7 +583,7 @@
  *		#define DATE_FORMAT "%a %b %d %T %Y"
  */
 static void
-strptime (char *ti, struct tm *tm)
+xstrptime (char *ti, struct tm *tm)
 {
 	/* Expect something like "Sat Mar 16 21:24:27 2013" */
 	static char *mon[] = { "Jan", "Feb", "Mar", "Apr", "May", "Jun",
@@ -625,8 +625,8 @@
 
 	gtk_tree_model_get(model, a, DATE_COLUMN, &time1, -1);
 	gtk_tree_model_get(model, b, DATE_COLUMN, &time2, -1);
-	strptime (time1, &tm1);
-	strptime (time2, &tm2);
+	xstrptime (time1, &tm1);
+	xstrptime (time2, &tm2);
 	t1 = mktime (&tm1);
 	t2 = mktime (&tm2);
 
